src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/ComInterfaceGenerator.cs

```diff
@@ -185,6 +185,18 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                     using StringWriter source = new();
                     source.WriteLine("// <auto-generated />");
                     source.WriteLine("#pragma warning disable CS0612, CS0618"); // Suppress warnings about [Obsolete] member usage in generated code.
+
+                    // If the user has specified 'ManagedObjectWrapper', it means that the COM interface will never be used to marshal a native
+                    // object as an RCW (eg. the IDIC vtable will also not be generated, nor any additional supporting code). To reduce binary
+                    // size, we're not emitting the interface methods on the implementation interface that has '[DynamicInterfaceCastableImplementation]'
+                    // on it. However, doing so will cause the CA2256 warning to be produced. We can't remove the attribute, as that would cause
+                    // the wrong exception to be thrown when trying an IDIC cast with this interface (not 'InvalidCastException'). Because this is
+                    // a niche scenario, and we don't want to regress perf or size, we can just disable the warning instead.
+                    if (interfaceContext.Options is ComInterfaceOptions.ManagedObjectWrapper)
+                    {
+                        source.WriteLine("#pragma warning disable CA2256");
```

Comment:
@MichalStrehovsky sanity check: should we be worried about ILC potentially emitting warning at publish time if we skip implementing methods on some IDIC implementation type? Also does 'IlcResilient' affect this at all?

Reply:
No, this is purely a run-time issue. We don't have a need to detect this at compile time so ILC doesn't check. You'll get the same `EntryPointNotFoundException` at the time of dispatch as with a JIT.

Reply:
Perfect, thank you for confirming! 🙂